src/App/Header/index.tsx

```diff
@@ -1,68 +1,33 @@
-import { DappLogo } from "components/Image";
 import { appRoutes } from "constants/routes";
-import { useEffect, useRef, useState } from "react";
-import { Location, matchRoutes, useLocation } from "react-router-dom";
+import { useLocation } from "react-router-dom";
 import { Link } from "../types";
-import DesktopNav from "./DesktopNav";
-import { Opener as MobileNavOpener } from "./MobileNav";
+import Logo from "./Logo";
+import NavDropdown from "./NavDropdown";
 import UserMenu from "./UserMenu";
+import useHeaderClassNames from "./useHeaderClassNames";
+
+const HEADER_ID = "navbar";
 
 type Props = { classes: string; links: Link[] };
 
 export default function Header({ classes, links }: Props) {
   const location = useLocation();
-  const isScrolledRef = useRef<boolean>(false);
-  const [isScrolled, setIsScrolled] = useState(false);
 
-  useEffect(() => {
-    function handleScroll() {
-      const _isScrolled = window.scrollY > 0;
-      if (_isScrolled !== isScrolledRef.current) {
-        setIsScrolled(_isScrolled);
-        isScrolledRef.current = _isScrolled;
-      }
-    }
-    window.addEventListener("scroll", handleScroll, { passive: true });
-    return () => {
-      window.removeEventListener("scroll", handleScroll);
-    };
-  }, []);
+  const headerClassNames = useHeaderClassNames(HEADER_ID);
 
-  const bg =
-    !hasBanner(location) || isScrolled ? "bg-white dark:bg-blue-d3" : "";
   return (
-    <header
-      className={`${classes} ${isScrolled ? "shadow-lg" : ""} ${bg}
-    transition-shadow ease-in-out duration-300 w-full h-[90px] mb-0`}
-    >
-      <div className="grid items-center gap-4 px-5 grid-cols-[auto_1fr_auto] h-full bg-white">
-        <DappLogo />
-        <DesktopNav
-          classes="hidden lg:flex font-heading font-bold uppercase"
-          links={links}
-        />
-        <div className="flex gap-4 justify-self-end items-center">
+    <header id={HEADER_ID} className={`${classes} ${headerClassNames}`}>
+      <div className="grid place-items-center gap-4 pl-5 pr-10 grid-cols-3 h-full w-full max-w-6xl mx-auto rounded-full bg-white">
```

Comment:
@SovereignAndrey @misicnenad, is redesign  a total package for RC-v1.9.2 ? - which warrants temporarily shipping to staging inconsistent UIs
<img width="1470" alt="image" src="https://github.com/AngelProtocolFinance/angelprotocol-web-app/assets/89639563/0ed89e81-9681-4369-9794-60b1a7944ea7">

Perhaps we ship incrementally e.g. , retain the current header layout (full width rectangle) but only update its contents 
like `login/ signup btns` dropdown links
<img width="401" alt="image" src="https://github.com/AngelProtocolFinance/angelprotocol-web-app/assets/89639563/2e8139ba-64d8-4a1e-a5c8-bfcce994a6d8">


The totality of this new header 
also requires blue background which is not the case on every page of our app (e.g. admin has white background)

other UI inconsistency would be corner roundness - most of our UI is `rounded` while we are introduction some `rounded-full`. Perhaps these could be in batches 
task 1: update theme colors 
task 2: update container roundness

Reply:
It shouldn't be a problem for the staging site to have inconsistent UI, but it would be a problem to have this in production.

If we wanted to keep staging site "consistent", we could create a special branch to hold all the design-related changes; that way we can merge UI rework in a batch. The downside would be that we would be setting ourselves up for some long conflict resolution sessions when trying to merge this new branch into `RC-v1.9.2`, which would still receive non-UI related updates.

----
**@SovereignAndrey @ap-justin what do you think of a completely new convention for how we organize our branches:**
- **`master` -> branch that holds production-ready code, the one accessible on the production site**
- **`staging` -> branch that holds staging code, accessible on `https://staging.better.giving/`**
- **RC-v1.9.2` -> development branch, changes that are not yet ready for the staging site.**

Changes would be merged into the "upper layer" branch in logical batches in the following order:
`RC-v1.9.2 -> staging -> master`

Merging would happen only when we are completely sure that the current batch (of changes) is ready to move to the next "layer".

Reply:
I've been thinking about this some more and I think the proposed branch organization (the latter one) could actually complicate things a bit. In our particular case, we'd then need to spin up a new "development" webapp stage, where everyone could access the in-development webapp - this is actually what we use "staging" for atm.
In short, this would only be adding one more step to the overall development and deployment processes.

The original proposal (having a `ui-rework` branch to hold all UI changes) is much more useful IMHO. If we kept this branch up-to-date with the most recent `RC-v1.9.2` changes, then there wouldn't be that many conflicts to solve; the only thing to note here is that *all UI-related changes* (**all of them**) would need to come from and into this `ui-rework` branch.

Reply:
@misicnenad @ap-justin Will try to answer all the questions and provide feedback:
1. UI Redesign IS the total package for RC-v1.9.2 shipping (at this point) and what launch will be waiting on along with wrapping up multiple donations and the new payment options.
2. As for branch use/naming proposals here's my take:
  - `master` -> branch that holds production-ready code
  - `RC-vX.X.X` -> Release Candidate branch that holds staging server version of code
  - `FB-<name>` -> Feature branches that hold (1 or more merged in working branches). These are the branches that we can build and refine with multiple, smaller PRs to progress a feature along until we're all happy enough with everything and know with some confidence that it won't terribly break staging (or cause Operations folks to freak out over known bugs).
   - `<working-branch>` -> linked to Linear tickets and flow into a FB (for bigger items like a UI overhaul) OR into a RC branch for smaller, one-off issues or features.

Reply:
Do we want to create this `FB-<name>` branch for the UI rework?
Or merge this into `RC-v1.9.2`?

Reply:
I feel that, since we've pushed a ton of re-design work into the `RC-v1.9.2` branch already, we can continue to do so. We'll take a `master < RC < FB < WB` approach for the next bigger features, in future releases, that would warrant that approach (Gift Cards, Leaderboard, Admin area improvements, etc).

Reply:
Will open the PR for review then and resolve this conversation.

Reply:
> I feel that, since we've pushed a ton of re-design work into the RC-v1.9.2 branch already, we can continue to do so. 

Yup, I'm only suggesting a more strategic way to do it so that internal testing team which constantly looking at staging won't create tickets like `Hey, staging nav bar is currently broken` - and could potentially hang in there at the moment while follow up works is being done. We could keep this confidence that what's up there is being evaluated for production and that bugs/inconsistencies are not liberally put up. 

As an example, I didn't settle on adding the [donate `Cancel` button](https://linear.app/better-giving/issue/BG-1104/move-cancel-button-at-the-top-of-the-page) without updating the layout alongside as it is out of place if I would just put it up there on the corner. 

In case of this navbar - shipping bit by bit is not that problematic: even so makes PRs even smaller. 
1.  PR1: inside header, update auth buttons, move links to dropdown
2. PR2: make blue background common to pages, make it navbar floating (not full width)
3. PR3: apply change in roundess of containers across app
4. etc..